.github/workflows/CryptatorTest.yml

```diff
@@ -12,5 +12,12 @@ jobs :
       java-version: '11'
       distribution: 'adopt'
       cache: 'maven'
+   - name: Cache local Maven repository
+     uses: actions/cache@v3
+     with:
+      path: ~/.m2/repository
+      key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+      restore-keys: |
+       ${{ runner.os }}-maven-
    - name: Test with Maven
-     run: mvn -B test -Dtest=ExtensiveTesting --file pom.xml
+     run: mvn test --file pom.xml; mvn test -Dtest=ExtensiveTesting --file pom.xml
```

Comment:
Are you sure that the action will fail if the classic test fails.
A shell script would not fail.  

Isn't it simpler/safer to create two distinct steps ?